src/darwin/Framework/CHIP/MTROTAProviderDelegate.h

```diff
@@ -58,6 +58,11 @@ typedef void (^MTRApplyUpdateRequestCompletionHandlerDeprecated)(
  * an error response to the client.  Otherwise it must have a non-nil data,
  * which will be returned to the client.
  */
+- (void)handleQueryImageForNodeID:(NSNumber *)nodeID
+                       controller:(MTRDeviceController *)controller
+                           params:(MTROTASoftwareUpdateProviderClusterQueryImageParams *)params
+            canStartNewBDXSession:(bool)canStartNewBDXSession
+                       completion:(MTRQueryImageCompletionHandler)completion MTR_NEWLY_AVAILABLE;
```

Comment:
Why we not just changing the existing `handleQueryImageForNodeID:controller:parans:completion:` as opposed to adding a new thing alongside it?

Also, this needs documentation explaining what the new boolean means, etc.

Reply:
i was not sure how we are doing backward compatibility. This is on me. I will fix it.

Reply:
will add documentation

Reply:
> i was not sure how we are doing backward compatibility

Short story: anything marked MTR_NEWLY_AVAILABLE has not shipped anywhere yet and can be changed.  Anything else cannot be changed.  ;)